### PR TITLE
[assimp] update to 5.4.0

### PR DESCRIPTION
**Pull request by miyanyan**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/assimp/build_fixes.patch [hunk 1]
```diff
@@ -1,520 +1,503 @@
...
+diff --git a/code/AssetLib/Blender/BlenderTessellator.h b/code/AssetLib/Blender/BlenderTessellator.h
+index e43535f..ce7340b 100644
+--- a/code/AssetLib/Blender/BlenderTessellator.h
++++ b/code/AssetLib/Blender/BlenderTessellator.h
+@@ -143,7 +143,7 @@ namespace Assimp
+ 
+ #if ASSIMP_BLEND_WITH_POLY_2_TRI
+ 
+-#include "contrib/poly2tri/poly2tri/poly2tri.h"
++#include "poly2tri/poly2tri.h"
```
Comment (Osyotr) [inline ports/assimp/build_fixes.patch:118]: Why change that? (ditto below)

File: ports/assimp/build_fixes.patch [hunk 2]
```diff
@@ -1,520 +1,503 @@
...
+diff --git a/code/AssetLib/MMD/MMDPmxParser.cpp b/code/AssetLib/MMD/MMDPmxParser.cpp
+index 5a3e61d..f2ac29e 100644
+--- a/code/AssetLib/MMD/MMDPmxParser.cpp
++++ b/code/AssetLib/MMD/MMDPmxParser.cpp
+@@ -42,7 +42,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ #include <utility>
+ #include "MMDPmxParser.h"
+ #include <assimp/StringUtils.h>
+-#include "utf8.h"
++#include "utf8cpp/utf8.h"
```
Comment (Osyotr) [inline ports/assimp/build_fixes.patch:174]: The proper usage for this library is `#include <utf8.h>`. (ditto below)

Comment (miyanyan) [reply ports/assimp/build_fixes.patch:174]: Hi, I changed the build_fixes.patch based on it's previous version, so I just changed the git diff lines but not the actual content。
I can change all the `#include "utf8cpp/utf8.h"`

**Comment by miyanyan** [comment]
closed, see https://github.com/microsoft/vcpkg/pull/38126

**Pull request closed without merging**